### ISSUE TITLE
Option (or filter API) to handle EOF

### ISSUE THREAD

**Issue by tobia** [hanslub42/rlwrap#97]
Hello

I'm trying to use rlwrap to "tame" a really good REPL, that unfortunately had its origins in Windows land (as well as not being open source) and therefore does not comply with UNIX conventions. In particular, Ctrl-D causes it to barf an error and keep running.

I looked into rlwrap options, as well as its filter API, for a way to intercept EOF coming from the user and translate it into a custom exit command (which would be `)off` in my case) but I couldn't find a way to do it.

**Comment by hanslub42** [comment]
This is already possible:

First, convince your terminal to _not_ interpret `CTRL+D` as a control character:

    $ stty eof undef

Then, bind `CTRL-D` to the text `off` + a newline in your `.inputrc`:

    $if some_repl
       "\C-d" : "off\n"
    $endif


Finally, run your REPL within `rlwrap`, stlil within the same terminal window:

    $ rlwrap some_repl

Pressing `CTRL+D` will then send the command `off` to your REPL.

**Comment by simaoafonso-pwt** [comment]
Sorry to come back to this, but I tried this it doesn't work. I changed the `stty` on my local terminal, changed my local inputrc, but the target program gets `\n\4\n`, from what I can tell.
When rlwrap terminates, the terminal is keeps the `stty` setting, whihc means it will stop interpreting `CTRL-d` on other programs, right? Can't rlwrap save and restore this terminal setting, doesn't it save and restore others already?
